src/core/server/saved_objects/migrations/model/retry_state.ts

```diff
@@ -12,13 +12,18 @@ export const delayRetryState = <S extends State>(
   state: S,
   errorMessage: string,
   /** How many times to retry a step that fails */
-  maxRetryAttempts: number
+  maxRetryAttempts: number,
+  /** optional link to docs */
+  docLink?: string
 ): S => {
+  const defaultReason = `Unable to complete the ${state.controlState} step after ${maxRetryAttempts} attempts, terminating.`;
   if (state.retryCount >= maxRetryAttempts) {
     return {
       ...state,
       controlState: 'FATAL',
-      reason: `Unable to complete the ${state.controlState} step after ${maxRetryAttempts} attempts, terminating.`,
+      reason: docLink
```

Comment:
Cloud usually timeout on a Kibana upgrade before we exhaust all the retry attempts. So we should provide an actionable log entry the first time a index status isn't yellow because most cloud users won't see this last retry and the fatal state.

Reply:
The first log entry might be overlooked. Would it not be better to add the actionable log entry to every retry, just to be sure it can be found?

Also, should we add a link to the docs or a more verbose message similar to what the docs suggest? For example:
```
Inspect the output of the `_cluster/allocation/explain?index=${targetIndex}` for more information.
```

Reply:
@rudolf I've added a link to the docs in every log message: [`2ae943c` (#130352)](https://github.com/elastic/kibana/pull/130352/commits/2ae943ceba9e1251359c82fb9e56ba046094f0d9)

Thanks for the tip!

Reply:
> Would it not be better to add the actionable log entry to every retry, just to be sure it can be found?

Absolutely, that's what I had in mind but didn't communicate that very clearly.